include/alpaka/mem/buf/BufCpu.hpp

```diff
@@ -45,12 +45,12 @@ namespace alpaka
             static_assert(!std::is_const<TIdx>::value, "The idx type of the buffer can not be const!");
 
         public:
-            template<typename TExtent>
-            ALPAKA_FN_HOST BufCpuImpl(DevCpu const& dev, TExtent const& extent)
-                : AllocCpuAligned<std::integral_constant<std::size_t, core::vectorization::defaultAlignment>>()
-                , m_dev(dev)
+            template<typename TExtent, typename Deleter>
+            ALPAKA_FN_HOST BufCpuImpl(DevCpu const& dev, TElem* pMem, Deleter deleter, TExtent const& extent)
```

Comment:
Since you are stuffing the deleter into a `std::function` later anyway, I would just use it in the constructor interface as well. But that's just personal taste.
```suggestion
            template<typename TExtent>
            ALPAKA_FN_HOST BufCpuImpl(DevCpu const& dev, TElem* pMem, std::function<void(TElem*)> deleter, TExtent const& extent)
```

Reply:
Sure, I can make the change in the `BufCpuImpl` constructor.

I would not make the change for the `BufUniformCudaHipRt` constructor, since the latter passes the deleter to a `shared_ptr`, and I don't know how that's handled internally.

And I would not make the change for the `BufCpu` constructor, to keep the interface as close as possible among the different buffers.

Reply:
OK, fine for me!